nixos/modules/installer/tools/nixos-generate-config/src/lib.rs

```diff
@@ -0,0 +1,114 @@
+#[macro_use]
+extern crate simple_error;
+
+use std::error;
+use std::result;
+use std::string::String;
+use std::process::Command;
+
+const MAN_EXECUTABLE: &str = env!("MAN_EXECUTABLE");
+const NIXOS_MANPAGES: &str = env!("NIXOS_MANPAGES");
+
+type Result<T> = result::Result<T,Box<dyn error::Error>>;
+
+struct Options {
+    out_dir: String,
+    root_dir: String,
+    force: bool,
+    no_filesystems: bool,
+    show_hardware_config: bool,
+    show_help: bool
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Error {
+    err: String,
+}
+
+fn parse_options(args: &[String]) -> Result<Options> {
+    let mut i : usize = 0;
+    let mut opts = Options {
+        out_dir: String::from("/etc/nixos"),
+        root_dir: String::from("/"),
+        force: false,
+        no_filesystems: false,
+        show_hardware_config: false,
+        show_help: false,
+    };
+    loop {
```

Comment:
`clap` or if you are really against a lot of dependencies perhaps `getopts`? Nice thing about `clap` is that it can also generate shell completions for you, etc.

Reply:
Yeah, I thought about that. Maybe later right now I think it quite significantly adds libraries that we would not have otherwise. And we don't need that many options right now. The current variant is more or less bug compatible with the old code.